### PR TITLE
Rfe trello 1782 create demo test

### PR DESCRIPTION
**Pull request by talgoldy**

### CODE REVIEW

File: eyes.sdk.core/pom.xml [hunk 1]
```diff
@@ -23,31 +23,15 @@
     <dependencies>
         <dependency>
             <groupId>com.applitools</groupId>
-            <artifactId>eyes-common-java3</artifactId>
+            <artifactId>eyes-connectivity-java3-jersey2x</artifactId>
             <version>3.160.3</version>
```
Comment (itaibh) [inline eyes.sdk.core/pom.xml:27]: change the order of the dependencies back to how it used to be so we see less changes while reviewing

File: eyes.common.java/src/main/java/com/applitools/connectivity/ServerConnector.java [hunk 2]
```diff
@@ -0,0 +1,563 @@
...
+    private String apiKey = null;
+    private RenderingInfo renderingInfo;
+
+    /***
+     * @param restClient The client for communication
+     * @param logger    Logger instance.
+     * @param serverUrl The URI of the rest server.
+     */
+    public ServerConnector(HttpClient restClient, Logger logger, URI serverUrl) {
+        super(restClient, logger, serverUrl);
```
Comment (itaibh) [inline eyes.common.java/src/main/java/com/applitools/connectivity/ServerConnector.java:55]: Check if this file can be moved into `sdk.core` and let it create its own `HttpClient`.

File: eyes.selenium.java/src/main/java/com/applitools/eyes/selenium/rendering/VisualGridEyes.java [hunk 3]
```diff
@@ -595,19 +596,12 @@ public void onRenderFailed(Exception e) {
...
-
-            // We are interested in tests which should be opened or are open.
-            if (testIsOpenAndNotClosed || lastTaskIsNotAClosingTask)
-            {
+                VisualGridTask.TaskType taskType = visualGridTask.getType();
+                if ((taskType == null && test.isOpenTaskIssued() && !test.isCloseTaskIssued()) ||
+                        (taskType != null && taskType != VisualGridTask.TaskType.CLOSE && taskType != VisualGridTask.TaskType.ABORT)) {
+                    filteredTests.add(test);
+                }
+            } else if (!test.isCloseTaskIssued()) {
```
Comment (itaibh) [inline eyes.selenium.java/src/main/java/com/applitools/eyes/selenium/rendering/VisualGridEyes.java:604]: bring back the old code, it's the correct one.

File: eyes.selenium.java/pom.xml [hunk 4]
```diff
@@ -23,17 +23,19 @@
...
+            <artifactId>eyes-sdk-core-java3</artifactId>
             <version>3.160.3</version>
         </dependency>
         <dependency>
             <groupId>com.applitools</groupId>
             <artifactId>eyes-sdk-core-java3</artifactId>
             <version>3.160.3</version>
+            <type>test-jar</type>
+            <scope>test</scope>
         </dependency>
```
Comment (itaibh) [inline eyes.selenium.java/pom.xml:35]: remove this dependency

File: eyes.selenium.java/passedTestsSuite.xml [hunk 5]
```diff
@@ -285,4 +285,13 @@
...
     </test>
+    <test name="TestVisualGridMissingSteps" parallel="methods">
+        <classes>
+            <class name="com.applitools.eyes.renderingGrid.VisualGrid_NoMissingSteps">
+                <methods>
+                    <include name="TestNoMissingSteps" />
+                </methods>
+            </class>
+        </classes>
+    </test>
```
Comment (itaibh) [inline eyes.selenium.java/passedTestsSuite.xml:296]: remove this test

File: eyes.sdk.core/src/main/java/com/applitools/eyes/EyesBase.java [hunk 6]
```diff
@@ -304,6 +316,7 @@ public static URI getDefaultServerUrl() {
      */
     public void setLogHandler(LogHandler logHandler) {
         logger.setLogHandler(logHandler);
+        serverConnector.setLogger(logger);
     }
```
Comment (itaibh) [inline eyes.sdk.core/src/main/java/com/applitools/eyes/EyesBase.java:320]: init the `serverConnector` logger when creating the `ServerConnector` instance. Also make sure to update in `setLogger` (not to be mistaken with `setLogHandler`)

File: eyes.sdk.core/pom.xml (hunk 1 above)
Comment (talgoldy) [reply eyes.sdk.core/pom.xml:27]: done

File: eyes.selenium.java/src/main/java/com/applitools/eyes/selenium/rendering/VisualGridEyes.java (hunk 3 above)
Comment (talgoldy) [reply eyes.selenium.java/src/main/java/com/applitools/eyes/selenium/rendering/VisualGridEyes.java:604]: Done

File: eyes.selenium.java/pom.xml (hunk 4 above)
Comment (talgoldy) [reply eyes.selenium.java/pom.xml:35]: Done

File: eyes.selenium.java/passedTestsSuite.xml (hunk 5 above)
Comment (talgoldy) [reply eyes.selenium.java/passedTestsSuite.xml:296]: Done

File: eyes.sdk.core/src/main/java/com/applitools/eyes/EyesBase.java (hunk 6 above)
Comment (talgoldy) [reply eyes.sdk.core/src/main/java/com/applitools/eyes/EyesBase.java:320]: Done

**Pull request closed without merging**